frontend/src/employee-frontend/state/user.tsx

```diff
@@ -37,8 +48,29 @@ export const UserContextProvider = React.memo(function UserContextProvider({
 }: {
   children: React.JSX.Element
 }) {
+  const [unauthorizedApiCallDetected, setUnauthorizedApiCallDetected] =
+    useState(false)
   const { data: authStatus, refetch } = useQuery(authStatusQuery())
-  const refreshAuthStatus = useCallback(() => void refetch(), [refetch])
+
+  const refreshAuthStatus = useCallback(async () => {
+    const result = await refetch()
+    // Reset logout detection if user is logged in again
+    if (result.data?.loggedIn) {
+      setUnauthorizedApiCallDetected(false)
+    }
+    return result
+  }, [refetch])
+
+  useEffect(() => {
+    const eventListener = ((loginStatusEvent: LoginStatusChangeEvent) => {
+      loginStatusEvent.preventDefault()
+      setUnauthorizedApiCallDetected(!loginStatusEvent.loginStatus)
+    }) as EventListener
```

Comment:
Omat DOM-eventit kuuluu toteuttaa `CustomEvent`-rajapinnalla. `as`-castauksesta pääsee eroon näin:

```typescript
export class LoginStatusChangeEvent extends CustomEvent<boolean> {
  static name = 'loginstatuschange' as const

  constructor(loginStatus: boolean) {
    super(LoginStatusChangeEvent.name, {
      cancelable: true,
      detail: loginStatus  // <-- detail-kenttään voi laittaa oman payloadin
    })
  }
}

// Näin kerrotaan TypeScript-tyypityksille omasta event-tyypistä
declare global {
  interface WindowEventMap {
    [LoginStatusChangeEvent.name]: CustomEvent<boolean>
  }
}

/* ... */

const eventListener = (loginStatusEvent: LoginStatusChangeEvent) => {
  loginStatusEvent.preventDefault()
  setUnauthorizedApiCallDetected(!loginStatusEvent.detail)
}
window.addEventListener(LoginStatusChangeEvent.name, eventListener)
```

Reply:
Muutettu käyttämään `CustomEvent`iä